File: trailing_string.py
#! /usr/bin/python

"""
Trailing String
Description:

You are given two strings 'A' and 'B'. Print out a 1 if string 'B' occurs at
the end of string 'A'. Else a zero. 

"""
from sys import argv

fileName = argv[1]

input = open(fileName)

for line in input.readlines():
	if len(line) <=1:
		continue

	haystack, needle = line.split(',')
	
	if haystack.strip().endswith(needle.strip()):
		print 1
	else:
		print 0
		
		
input.close()

File: trailing_string.js
#!/usr/local/bin/node
'use strict';

var fs = require('fs');
var myArgv = process.argv;
var fileName = myArgv[2];
var buffer = fs.readFileSync(fileName, 'utf8');
var lines = buffer.split('\n');
lines.forEach(function (line) {
  if (line != null){
    let [haystack, needle] = line.split(',');
    //var input = line.split(',');
    //var haystack = input[0];
    //var needle = input[1].trim();
    if(haystack.endsWith(needle.trim())){
      console.log(1);
    }else {
      console.log(0);
    }
  }
});
